planet/src/main/webapp/js/headerRenderer.js

```diff
@@ -0,0 +1,224 @@
+}
+
+function renderBio(bio, userId) {
+    document.getElementById('bio').innerHTML = '';
+    if (bio !== undefined) {
+        document.getElementById('bio').innerText = bio;
+        return;
+    } 
+    if (userId !== currentUser.uid) {
+        document.getElementById('bio').innerText = 'No Bio';
+        return;
+    }
+    const addBioLink = document.createElement('a');
+    addBioLink.addEventListener('click', addNewBio);
+    addBioLink.innerText = 'Add Bio';
+    document.getElementById('bio').appendChild(addBioLink);
+}
+
+function addNewLocation() {
+    document.getElementById('location').classList.remove('valign-wrapper');
+    const inputFields = document.createElement('div');
+    // Create two input fields
+    const city = createLocationInput('city-input', 'City');
+    const country = createLocationInput('country-input', 'Country');
+    // Create a comma
+    const text = document.createElement('span');
+    text.style.marginRight = '5px';
+    text.innerText = ',';
+    inputFields.appendChild(city);
+    inputFields.appendChild(text);
+    inputFields.appendChild(country);
+    // Create the save button
+    const saveButton = createHeaderButton('Save', saveLocation);
+    // Create the cancel button
+    const cancelButton = createHeaderButton('Cancel', cancelLocationEditing);
+    // Create a button container
+    const buttonsContainer = document.createElement('div');
+    buttonsContainer.classList.add('row');
+    buttonsContainer.style.margin = '0';
+    buttonsContainer.appendChild(saveButton);
+    buttonsContainer.appendChild(cancelButton);
+    // Render the inputs
+    document.getElementById('location').innerHTML = '';
+    document.getElementById('location').appendChild(inputFields);
+    document.getElementById('location').appendChild(buttonsContainer);
+}
+
+function createLocationInput(inputId, placeHolder) {
+    const input = document.createElement('input');
+    input.placeholder = placeHolder;
+    input.type = 'text';
+    input.style.borderRadius = '3px';
+    input.style.backgroundColor = 'white';
+    input.style.height = 'auto';
+    input.style.width = '30%';
+    input.style.margin = '0';
+    input.id = inputId;
+    return input;
+}
+
+function saveLocation() {
+    const city = document.getElementById('city-input');
+    const country = document.getElementById('country-input');
+    if (city !== null && country !== null && 
+        city.value !== '' && country.value !== '') {
+        addLocationToDatabse(city.value + ', ' + country.value);
+        return;
+    }
+    cancelLocationEditing();
+}
+
+function cancelLocationEditing() {
+    renderLocation(undefined, currentUser.uid);
+}
+
+function addLocationToDatabse(location) {
+    const userReference = database.ref('users/' + currentUser.uid);
+    userReference.update({
+        location: location
+    }).then(() => {
+        renderLocation(location, currentUser.uid);
+    });
+}
+
+function addNewBio() {
```

Comment:
Generally, functions should be named using the verb that the function is doing, not the verb that the user is doing.

In this case, it looks like the function is opening the Add New Bio dialog. Maybe it could be called openAddNewBioDialog or openBioDialog?

Reply:
I changed it to "addNewBioForm".